.github/workflows/main.yml

```diff
@@ -78,8 +78,19 @@ jobs:
         docker run hello-world
         echo "***** Displaying Docker information..."
         docker info
-        echo "***** Configuring Docker for containerd image store..."
+        echo "***** Configuring Docker for containerd image store and builder keepStorage..."
         echo "{ \"features\": { \"containerd-snapshotter\": true }}" | sudo tee /etc/docker/daemon.json
+        echo $'{
+          "features": {
+            "containerd-snapshotter": true
+          },
+          "builder": {
+            "gc": {
+              "defaultKeepStorage": "20GB",
```

Comment:
Optional suggestion: 
I’m not sure what the optimal `defaultKeepStorage` value is, 
but it might help to log free disk space in GitHub CI at the start and end of the workflow. For example, add:

```yaml
- run: df -h
```

both at the beginning and the end of our docker build  job. 

Monitoring available space will help us choose the right cache size.
See: https://carlosbecker.com/posts/github-actions-disk-space/